tests/scripts/all.sh

```diff
@@ -886,6 +888,63 @@ clang_version() {
     fi
 }
 
+gcc_version() {
+    gcc="$1"
+    if command -v "$gcc" > /dev/null ; then
+        "$gcc" --version | sed -En '1s/^[^ ]* \([^)]*\) ([0-9]+).*/\1/p'
+    else
+        echo 0  # report version 0 for "no gcc"
+    fi
+}
+
+can_run_cc_output() {
+    cc="$1"
+    result=1
+    if type "$cc" >/dev/null 2>&1; then
+        testbin=$(mktemp)
+        if echo 'int main(void){return 0;}' | "$cc" -o "$testbin" -x c -; then
+            if "$testbin" 2>/dev/null; then
+                result=0
+            fi
+        fi
+        rm "$testbin"
```

Comment:
this should probably be `rm -f`, in case `cc` doesn't work, in which case there won't be an output file, and then we'd get an error message without `-f`

Reply:
Hmm, I was assuming that cc truncates / replaces the file created by mktemp atomically - but I guess it's possible it could fail during that operation?

Reply:
A quick `strace` shows that the output file gets unlinked then re-created, so while unlikely it's always possible that there isn't one at the end. It's really just a question of script hygiene with `rm -f` for temporary files